react-hook/cra-implementation/src/index.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useRef, useState } from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const DecimalClock = () => {
+  const frameRequest = useRef(null);
+  const [date, setDate] = useState(new Date());
+  const [fractionOfDay, setFractionOfDay] = useState(null);
+  const [timeString, setTimeString] = useState(null);
+
+  const update = () => {
+    setDate(new Date());
+    frameRequest.current = requestAnimationFrame(update);
+  };
+
+  useEffect(() => {
+    frameRequest.current = requestAnimationFrame(update);
+
+    return () => {
+      cancelAnimationFrame(frameRequest.current);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const timeOfDay =
+      (date.getHours() * 3600 + date.getMinutes() * 60 + date.getSeconds()) *
+        1000 +
+      date.getMilliseconds();
+    setFractionOfDay(timeOfDay / 86400_000);
+  }, [date]);
```

Comment:
Calculation can be inlined. No need to trigger extraneous effects, state mutations, and renders.